src/main/java/Duke.java

```diff
@@ -1,10 +1,189 @@
+            } catch(FileNotFoundException e) {
+                System.out.println(e.getMessage());
+            }
+
+        }
+        String userInput = "";
+        System.out.println("Hello, What Can I do for you ?\n -------------------------------");
+        Scanner scanner = new Scanner(System.in);
+        while(!userInput.equals("bye")) {
+            System.out.println("Enter Input Here: ");
+
+            userInput = scanner.nextLine();
+            if (userInput.equals("bye")) {
+                System.out.println("Duke : Bye, Hope to see you again soon !");
+                break;
+            } else if (userInput.equals("list")) {
+                int numberOfTasks = tasks.size();
+                for (int i = 0; i < numberOfTasks; i++) {
+                    System.out.println(i + 1 + ". " + tasks.get(i));
+                }
+                System.out.println("\n ----------------------------------");
+
+            } else if (userInput.startsWith("done")) {
+                int id = Integer.valueOf(userInput.substring(5));
+                Task currTask = null;
+                try {
+                    currTask = tasks.get(id - 1);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println(e.getMessage());
+                }
+                if (currTask != null) {
+                    currTask.markAsDone();
+                    System.out.println("Nice! I've marked this task as done: \n" + currTask);
+                    System.out.println("\n ----------------------------------");
+                }
+
+            } else if (userInput.startsWith("delete")) {
+                int id = Integer.valueOf(userInput.substring(7));
+
+                try {
+                    Task removedTask = tasks.get(id - 1);
+                    tasks.remove(id);
+                    System.out.println("Successfully removed task : " + removedTask);
+                } catch (Exception e) {
+                    System.out.println(e.getMessage());
+                }
+            } else {
+                Task newTask = null;
+
+                try {
+                    newTask = handleInput(userInput);
+                } catch (DukeException duke) {
+                    System.out.println(duke.getMessage());
+                }
+                if (newTask != null) {
+                    tasks.add(newTask);
+                    System.out.println("Duke: Added Task " + userInput);
+
+                }
+                System.out.println("\n ----------------------------------");
+            }
+            try{
+                String tasksinstring = tasks.stream().map(x -> x.toString()).reduce("",
+                        (x,y) -> x + "\n" + y);
+                Files.write(taskListPath, tasksinstring.getBytes());
+            } catch(IOException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+    public static Task handleInput(String userInput) throws DukeException{
```

Comment:
Could add a space after DukeException

Reply:
Would be better if 139 is left as a blank line, and this method signature is written on the 140th line instead.
Try `DukeException {` instead of `DukeException{` to better adhere to the style guide.